packages/core/src/common/keybinding.ts

```diff
@@ -204,4 +232,52 @@ export class KeybindingRegistry {
         }
     }
 
+    setKeymap(rawKeyBindings: RawKeybinding[]) {
+        /* Only assign the default keybindings once, so that they do not get overridden with further iterations */
+        if (!this.defaultKeyBindings || !this.defaultCommands) {
+            this.defaultKeyBindings = Object.assign({}, this.keybindings);
+            this.defaultCommands = Object.assign({}, this.commands);
+        }
+        let invalidKeyMap: boolean = false;
+        for (const rawKeyBinding of rawKeyBindings) {
+            if (this.commandRegistry.getCommand(rawKeyBinding.command)) {
```

Comment:
This code will modify the previous map that is a mixture of defaultbindings and previous keymaps.